src/Math/Math.jl

```diff
@@ -131,11 +131,12 @@ end
 #@inline levin_scale(B::T, n, k) where T = -(B + n) * (B + n + k)^(k - one(T)) / (B + n + k + one(T))^k
 @inline levin_scale(B::T, n, k) where T = -(B + n + k) * (B + n + k - 1) / ((B + n + 2k) * (B + n + 2k - 1))
 
-@inline @generated function levin_transform(s::NTuple{N, T}, w::NTuple{N, T}) where {N, T <: FloatTypes}
+@inline @generated function levin_transform(s::NTuple{N, T}, 
+                                            w::NTuple{N, T}) where {N, T <: FloatTypes}
     len = N - 1
     :(
         begin
-            @nexprs $N i -> a_{i} = Vec{2, T}((s[i] * w[i], w[i]))
+            @nexprs $N i -> a_{i} = Vec{2, T}((s[i] / w[i], 1 / w[i]))
```

Comment:
So the rule could actually be simplified to...
```julia
function EnzymeRules.forward(func::Const{typeof(levin_transform)}, 
                               ::Type{<:Duplicated}, 
                               s::Duplicated,
                               w::Duplicated)
    (sv, dv, N) = (s.val, s.dval, length(s.val))
    ls  = levin_transform(sv, w.val)
    dls = levin_transform(dv, w.dval)
    Duplicated(ls, dls)
  end
```

If this line was changed to 
```julia
            @nexprs $N i -> a_{i} = isinf(w[i]) ? (return s[i]) : Vec{2, T}((s[i] * w[i], w[i]))
```

Which should return the correct scalar value when not called with enzyme as well as return correct value when using enzyme.

There's going to be some confusion because I was operating under the we are feeding in the 1 / weights instead. So this won't quite match up

Reply:
Excellent point. Will commit up very shortly.

Reply:
pushed up!